magnum_cluster_api/driver.py

```diff
@@ -220,6 +220,25 @@ def update_cluster_status(
                     cluster.save()
                     return
 
+            # Check reason if an individual machine is not ready
+            machines = objects.Machine.objects(self.k8s_api).filter(
+                namespace="magnum-system",
+                selector={
+                    "cluster.x-k8s.io/cluster-name": cluster.stack_id,
+                },
+            )
+
+            for machine in machines:
+                for cond in machine.obj["status"]["conditions"]:
+                    if (
+                        cond.get("type") == "InfrastructureReady"
+                        and cond.get("status") == "False"
+                    ):
+                        messagetext = cond.get("message")
+                        if messagetext:
+                            cluster.status_reason = messagetext
+                            cluster.save()
+
```

Comment:
This will overwrite `status_reason` generated by `_get_cluster_status_reason` in the former step.
Current `_get_cluster_status_reason` generates status reason message from `Cluster` and `OpenstackCluster` CRs. 
I recommend appending your code snippet into `_get_cluster_status_reason` function.
You'll need to loop all machines and concatenate messages with machine name. Then append this all-machine message in the final reason message alongside existing ones.

https://github.com/vexxhost/magnum-cluster-api/blob/b2a612f0b7aa6d71d2d390217d443f0f84695f25/magnum_cluster_api/driver.py#L124-L148

Reply:
Unfortunately the original author of this patch has now taken retirement and no longer works on my team. Realistically it is going to need someone else to pick this up as I don't have any engineering effort available to work on this for the foreseeable future.